File: src/main/java/com/pei/dehaze/model/form/DatasetForm.java
package com.pei.dehaze.model.form;

import com.pei.dehaze.common.validator.PathExists;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Schema(description = "数据集表单对象")
@Data
public class DatasetForm {
    @Schema(description = "数据集ID")
    private Long id;

    @Schema(description = "父数据集ID")
    @NotNull(message = "父数据集ID不能为空")
    private Long parentId;

    @Schema(description = "数据集类型")
    private String type;

    @Schema(description = "数据集名称")
    private String name;

    @Schema(description = "数据集描述")
    private String description;

    @Schema(description = "数据集存储路径")
    @PathExists
    private String path;

    @Schema(description = "状态(1:正常;0:禁用)")
    private Integer status;
}

File: dehaze-go/internal/model/read/dataset_form.go
package read

import "time"

// DatasetForm 数据集表单读模型
type DatasetForm struct {
	ID          *int64     `json:"id"`
	ParentID    int64      `json:"parentId"`
	Type        string     `json:"type"`
	Name        string     `json:"name"`
	Description string     `json:"description"`
	Path        string     `json:"path"`
	Status      int8       `json:"status"`
	CreateTime  time.Time  `json:"createTime"`
	UpdateTime  time.Time  `json:"updateTime"`
}
